src/umpire/strategy/DynamicPool.cpp

```diff
@@ -356,15 +374,14 @@ void DynamicPool::mergeFreeBlocks()
     std::tie(bytes, is_head, whole_bytes) = rec.second;
     insertFree(ptr, bytes, is_head, whole_bytes);
   }
+
+  UMPIRE_REPLAY("() Free blocks after: " << getFreeBlocks());
 }
 
 std::size_t DynamicPool::releaseFreeBlocks()
 {
   UMPIRE_LOG(Debug, "()");
 
-  // Coalesce first so that we are able to release the most memory possible
```

Comment:
Why is this being removed from here and how does it pertain to the fix?

Reply:
This was getting called twice through `coalesce`. `releaseFreeBlocks` should literally only release free blocks.

Reply:
This is kind of a slippery slope to me.  I now understand why `::allocateFromResource()` was calling `::releaseFreeBlocks()` instead of `::release()`.

If possible, I would like to avoid general clean-up in a patch release.